### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by turankaanelgin** [turankaanelgin/Action-Recognition#1]
I checked some papers which use bag-of-words model and they indicate that since the data are videos, the features should be motion-related. So instead of SIFT features, we have to use optical flow.

**Comment by turankaanelgin** [comment]
Without tuning, SVM predicts all training set as one label.

**Comment by vkhoi** [comment]
I checked through your code and tries to run it on my computer. I think there are some problems:
1. The "classifiers.values()" that you used on line 73 in bag_of_words.py does not return the classifiers in the order CATEGORIES. That's because python dictionary sorts their keys in a different order.
2. On line 60, if you call "examples += negatives", it will append negatives to histograms[category] (because examples is referencing to histograms[category]), which will modify the histograms and affect later SVM classifiers. So on line 55, you must call "examples = histograms[category].copy()" to create a copy. I've just tried that and the result is better.
